accfg/enable.c

```diff
@@ -101,13 +103,54 @@ static int device_action(int argc, const char **argv, const char *usage,
 		if (strcmp(argv[i], "all") == 0) {
 			argv[0] = "all";
 			argc = 1;
+			bmap_dev |= 0x3;
+			break;
+		}
+		if (strcmp(argv[i], "dsa") == 0) {
+			argv[0] = "dsa";
+			argc = 1;
+			bmap_dev |= 0x1;
+			break;
+		}
+		if (strcmp(argv[i], "iax") == 0) {
+			argv[0] = "iax";
+			argc = 1;
+			bmap_dev |= 0x2;
 			break;
 		}
 	}
 
-	for (i = 0; i < argc; i++) {
-		struct accfg_device *device;
+	if (bmap_dev) {
+		accfg_device_foreach(ctx, device) {
+			if (!accfg_device_is_active(device))
+				continue;
+
+			if (strstr(accfg_device_get_devname(device), "iax") &&
+				(bmap_dev & 0x2) == 0)
+				continue;
+			if (strstr(accfg_device_get_devname(device), "dsa") &&
+				(bmap_dev & 0x1) == 0)
+				continue;
+
+			rc = dev_action_switch(device, action);
+			if (rc == 0) {
```

Comment:
This entire if condition seems to be redundant. If dev_action_switch returned success, this should not be possible.

Can you please remove this check also in the original code below?